grafana-sync.py

```diff
@@ -333,6 +328,78 @@ def auth_keys(base_url, session):
     request.raise_for_status()
     print(request.json())
 
+class GrafanaLogin:
+    def __init__(self, token:str, user:str, password: str, base_url: str):
+        self.session = requests.Session()
+        if token:
+            self.session.headers = {"Authorization": f"Bearer {token}"}
+        else:
+            create_token(base_url, self.session, user=user, password=password)
+class GrafanaClient:
+    def __init__(self, base_url: str, session: GrafanaLogin, overwrite: bool) -> None:
+        super().__init__()
+        self.url = base_url
+        self.session = session.session
+        self.folders = self.__load_folders()
+        self.alerts = self.__load_alerts()
+        self.overwrite = overwrite
+
+    def copy_alerts_from(self, source: 'GrafanaClient'):
```

Comment:
why is `'GrafanaClient'` in quotes here? is this not a type hint?

Reply:
Python can not reference to a class that is not already defined. You can take a look at google, this post [explain it](https://stackoverflow.com/questions/3877947/self-referencing-classes-in-python). You need to quote them to import them as a future import. That is the reason of the quotes, and it will not run if you remove them